src/server/service/parking-lot.ts

```diff
@@ -0,0 +1,64 @@
+import ParkingLotModel from '../models/parking-lot';
+import { UnexistentParkingLotID } from '../error/parking-lot.service';
+
+import type { ParkingLot } from '../models/parking-lot';
+
+export type CreateParkingLotDTO = {
+  name: string;
+};
+
+export type UpdateParkingLotDTO = {
+  id: string;
+  name?: string;
+};
+
+export interface IParkingLotService {
+  create(dto: CreateParkingLotDTO): Promise<ParkingLot>;
+  update(dto: UpdateParkingLotDTO): Promise<ParkingLot>;
+  remove(id: string): Promise<void>;
+  findById(id: string): Promise<ParkingLot | null>;
+}
+
+export default class ParkingLotService implements IParkingLotService {
+  async create(dto: CreateParkingLotDTO): Promise<ParkingLot> {
+    const parkingLot = new ParkingLotModel(dto);
+
+    await parkingLot.save();
+
+    return parkingLot;
+  }
+
+  async update(dto: UpdateParkingLotDTO): Promise<ParkingLot> {
+    const parkingLot = await ParkingLotModel.findById(dto.id);
+
+    if (!parkingLot) {
+      throw new UnexistentParkingLotID();
+    }
```

Comment:
Why not just throwing `UnexistentParkingLotID` on `findById`? This way you bubble up the error to the caller and this condition is removed.

Also, you would always have to check for `UnexistentParkingLotID` instead of checking for `null`. So we keep consistency on error handling.